cnf-certification-test/lifecycle/suite.go

```diff
@@ -85,13 +87,44 @@ var _ = ginkgo.Describe(common.LifecycleTestKey, func() {
 	}
 })
 
+func testPodTaintsTolerationHelper(put *v1.Pod) (bool, error) {
+	if put == nil {
+		return false, errors.New("invalid input, pod under test is nil")
+	}
+	for _, t := range put.Spec.Tolerations {
+		if t.Effect == v1.TaintEffectNoSchedule || t.Effect == v1.TaintEffectPreferNoSchedule ||
+			t.Effect == v1.TaintEffectNoExecute {
+			logrus.Error(fmt.Sprintf("pod %s:%s has toleration %s\n", put.Namespace, put.Name, t.Effect))
+			return false, nil
```

Comment:
You're returning on the first bad toleration found. Unless they're all mutually exclusive, I believe you should iterate through all of them, so you can show all the "bad" ones in the claim file so the customer can fix them altogether. I suggest to change the function to something like:
```
getBadTolerationsTaintEffectFromPod(pod *v1.Pod) []string {
    badEffects := []string{}
    for _, t := range put.Spec.Tolerations {
          if t.Effect == v1.TaintEffectNoSchedule || t.Effect == v1.TaintEffectPreferNoSchedule ||
	          t.Effect == v1.TaintEffectNoExecute {
	          badEffects = append(badEffects , t.Effect)
          }
    }
    return badEffects 
}
```
Then, in the pods loops, you can use check that the output slice is empty:
```
if badEffects := getBadTaintTolerationsEffectsFromPod(put); len(badEffects) > 0 {
    badPods = append...
    tnf.ClaimFilePrintf("%s has invalid taint tolerations effects %v", provider.PodToString(put), badEffects)
}
```

Reply:
will fix it